src/rebar_pkg_resource.erl

```diff
@@ -32,6 +31,28 @@
 %%==============================================================================
 %% Public API
 %%==============================================================================
+
+-spec init(rebar_state:t()) -> {ok, term()}.
+init(State) ->    
+    HexConfig=#{api_url := DefaultApiUrl,
+                repo_url := DefaultRepoUrl,
+                repo_public_key := DefaultRepoPublicKey,
+                repo_verify := DefaultRepoVerify} = hex_core:default_config(),
+    ApiUrl = rebar_state:get(State, hex_api_url, DefaultApiUrl),    
+    RepoUrl = rebar_state:get(State, hex_repo_url, 
+                              %% check legacy configuration variable for setting mirrors
+                              rebar_state:get(State, rebar_packages_cdn, DefaultRepoUrl)),
+    RepoPublicKey = rebar_state:get(State, hex_repo_public_key, DefaultRepoPublicKey),
+    RepoVerify = rebar_state:get(State, hex_repo_verify, DefaultRepoVerify),
+    {ok, Vsn} = application:get_key(rebar, vsn),
+    {ok, #{hex_config => HexConfig#{api_url => ApiUrl,
+                                    repo_url => RepoUrl,
+                                    repo_public_key => RepoPublicKey,
+                                    repo_verify => RepoVerify,
+                                    http_user_agent_fragment => 
+                                        <<"(rebar3/", (list_to_binary(Vsn))/binary, ") (httpc)">>,
+                                    http_adapter_config => #{profile => rebar}}}}.
```

Comment:
so we'll end up with one instance of the pkg resource per repo and currently there's only the default one or something?

Reply:
I was thinking pkg resource would learn how to handle multiple repos. So init would create a list of repos it knows about and the corresponding hex core config for them.

Reply:
Sounds good. I wasn't properly in the context of the first task which is just switching the one single repo to the new hex anyway

Reply:
Ah, yup. I figured I'd get this commit fixed up and merged into that new erlang/rebar3 branch. Then the changes to make it multi-repo will be a PR on that branch, making for a smaller set of changes that are easier to review.